src/test/java/store/service/SaleServiceTest.java

```diff
@@ -1,227 +1,282 @@
+        when(typedQuery.getResultList()).thenReturn(saleList);
+
+        List<SaleDTO> resultListDTO = saleService.get(ID, saleDateParameter, categoriesParameter);
+        //convert saleDTO into sale
+        List<Sale> resultList = resultListDTO.stream().map(saleDTO -> mapper.toEntity(saleDTO, Sale.class)).collect(Collectors.toList());
+
+        //testing the method is calling
+        verify(typedQuery).getResultList();
+
+        //testing correct data
+        assertEquals(saleList, resultList);
+        assertEquals(1, resultList.size());
+    }
+
+    @Test
+    public void testIfPassSaleDateParameter(){
+        //given
+        LocalDate date = LocalDate.of(YEAR, 4, DAY);
+
+        //when
+        when(entityManager.getCriteriaBuilder()).thenReturn(criteriaBuilder);
+        when(criteriaBuilder.createQuery(Sale.class)).thenReturn(saleCriteriaQuery);
+        when(saleCriteriaQuery.from(Sale.class)).thenReturn(saleRoot);
+        when(saleRoot.get(Sale_.id)).thenReturn(path);
+        when(entityManager.createQuery(saleCriteriaQuery)).thenReturn(typedQuery);
+        when(typedQuery.getResultList()).thenReturn(saleList);
+
+        //then
+        List<SaleDTO> resultListDTO = saleService.get(idParameter, date, categoriesParameter);
+        List<Sale> resultList = resultListDTO.stream().map(saleDTO -> mapper.toEntity(saleDTO, Sale.class)).collect(Collectors.toList());
+
+        //testing the method is calling
+        verify(typedQuery).getResultList();
+
+        //checking correct data
+        assertEquals(saleList, resultList);
+        assertEquals(1, resultList.size());
+    }
+
+    @Test
+    public void testIfPassCategoriesParameter(){
+        //given
+        List<String> cat = Arrays.asList("First", "Second");
+
+        //when
+        when(entityManager.getCriteriaBuilder()).thenReturn(criteriaBuilder);
+        when(criteriaBuilder.createQuery(Sale.class)).thenReturn(saleCriteriaQuery);
+        when(saleCriteriaQuery.from(Sale.class)).thenReturn(saleRoot);
+        when(saleRoot.get(Sale_.id)).thenReturn(path);
+        when(entityManager.createQuery(saleCriteriaQuery)).thenReturn(typedQuery);
+        when(saleRoot.join(anyString())).thenReturn(joinCat);
+        when(typedQuery.getResultList()).thenReturn(saleList);
+
+        //then
+        List<SaleDTO> resultListDTO = saleService.get(idParameter, saleDateParameter, cat);
+        List<Sale> resultList = resultListDTO.stream().map(saleDTO -> mapper.toEntity(saleDTO, Sale.class)).collect(Collectors.toList());
+
+        //testing the method is calling
+        verify(typedQuery).getResultList();
+
+        //checking correct data
+        assertEquals(saleList, resultList);
+        assertEquals(1, resultList.size());
+    }
+
+
+
+    @Test
+    public void testAdd(){
+        //given
+        LocalDate dateOn = LocalDate.of(YEAR, 1, DAY);
+        LocalDate dateOff = LocalDate.of(YEAR, 2, DAY);
+        Sale newSale = new Sale();
```

Comment:
you have already had static variable "sale" . I think it would be better to use it instead of creating new object